enhancements/cluster-logging/cluster-logging-es-pipeline-processing.md

```diff
@@ -0,0 +1,91 @@
+
+## Summary
+
+This proposal opens possibility to route individual documents (logs) into different Elasticsearch indices
+based on data driven criteria.  Under the hood it utilizes Elasticsearch Ingesting [Pipeline Definition](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/pipeline.html) to dynamically inspect incoming documents and apply appropriate processor(s) such as [parsing JSON data](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/json-processor.html) in document field or pointing
+the document to a different index (similarly to [date-index-name-processor](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/date-index-name-processor.html)).
+
+This will enable indexing and storing of documents that carry custom data (such as JSON payload) into different index to avoid index mapping collisions and improves Elasticsearch querying and Kibana dashboarding experience.
+
+## Motivation
+
+One of the goals of [introduction of the new data model](cluster-logging-es-rollover-data-design.md) was to significantly reduce the number of indices in Elasticsearch cluster. Keeping the number of indices (and thus index shards) at bay is a critical step towards improved Elasticsearch stability and performance.
+
+We had to make sure we could switch from _"index per namespace model"_ to _"common shared index for all projects"_ (see the new data proposal for details). In order to store the data in common index Elasticsearch requires the data must share the same data model which directly translates to specific [index mappings](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/mapping.html). Data having conflicting model can not be stored in common index.
+
+However, common data model does not solve another challenge associated with the ask to properly index, store and query any additional (extra) custom data the documents can carry. For example logs can contain project or business domain specific JSON data that is not captured in the common data model and is not shared by all the data sources.
+
+Over the time we have gathered number of requests from customers asking for possibility to process additional data. See <https://issues.redhat.com/browse/LOG-785>.
+
+For the purpose of this proposal we have alerady investigated various approaches (and their pros and cons) of storing additional JSON data in Elasticsearch indices as part of <https://issues.redhat.com/browse/LOG-842>.
+
+
+### Goals
+
+The high-level goal of this proposal is to lay the groundwork to enable approaches B) and C) as discussed in [LOG-842](https://issues.redhat.com/browse/LOG-842). Depending on how much users are willing to pay attention to their data model they can start with approach B) (which requires less data model attention but is more system resource intensive) and eventually transition to approach C) once they feel more confident about managing their own data model across more log resources.
```

Comment:
I think its worth at least summarizing B and C since this enhancement would ultimately be the source of truth and others may not have access to JIRA

Reply:
agreed -- that should be part of the "proposal" body

Reply:
@jcantrill Done. Hope it is more clear now.